<h1> Palmer Penguin Data Analytics </h1>
<hr>
<h3>Analyst:Sharlene Mae Habito</h3>

In [1]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

<p>Importing dataset to Jupyter notebook</p>
<p>Converting dataset to dataframe</p>
<ul> Requirements
    <li> Name of the dataframe</li>
    <li> Source </li>
</ul>

In [2]:
penguins = pd.read_csv('datasets\\penguins.csv')

In [3]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h3>Descriptive Analytics</h3>

In [4]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [5]:
penguins.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

<h3>Cleaning the Data Frame</h3>

In [6]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h3>Selecting a column</h3>
<h3>Syntax</h3>

data_frame['column_name']

In [7]:
penguins['island']

0      Torgersen
1      Torgersen
2      Torgersen
3      Torgersen
4      Torgersen
         ...    
339        Dream
340        Dream
341        Dream
342        Dream
343        Dream
Name: island, Length: 344, dtype: object

In [8]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [9]:
penguins['species'].unique()

array(['Adelie', 'Gentoo', 'Chinstrap'], dtype=object)

In [10]:
Adelie_df = penguins[penguins['species'] =='Adelie']

In [11]:
Adelie_df

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[152 rows x 8 columns]

In [12]:
Gentoo_df = penguins[penguins['species'] =='Gentoo']

In [13]:
Gentoo_df

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN            NaN                NaN   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 8 columns]

In [14]:
chinstrap_df = penguins[penguins['species'] =='Chinstrap']

In [15]:
chinstrap_df

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [16]:
penguins[penguins['body_mass_g'] >= 4000]

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
7       Adelie  Torgersen            39.2           19.6              195.0   
9       Adelie  Torgersen            42.0           20.2              190.0   
14      Adelie  Torgersen            34.6           21.1              198.0   
17      Adelie  Torgersen            42.5           20.7              197.0   
19      Adelie  Torgersen            46.0           21.5              194.0   
..         ...        ...             ...            ...                ...   
323  Chinstrap      Dream            49.0           19.6              212.0   
329  Chinstrap      Dream            50.7           19.7              203.0   
333  Chinstrap      Dream            49.3           19.9              203.0   
339  Chinstrap      Dream            55.8           19.8              207.0   
342  Chinstrap      Dream            50.8           19.0              210.0   

     body_mass_g   sex  year  
7         4675.0  male  2007  
9         4250.0   NaN  2007  
14        4400.0  male  2007  
17        4500.0  male  2007  
19        4200.0  male  2007  
..           ...   ...   ...  
323       4300.0  male  2009  
329       4050.0  male  2009  
333       4050.0  male  2009  
339       4000.0  male  2009  
342       4100.0  male  2009  

[177 rows x 8 columns]

In [17]:
penguins[(penguins['species'] == "Chinstrap") & (penguins['body_mass_g'] >=4000)].info()

<class 'pandas.core.frame.DataFrame'>
Index: 16 entries, 284 to 342
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            16 non-null     object 
 1   island             16 non-null     object 
 2   bill_length_mm     16 non-null     float64
 3   bill_depth_mm      16 non-null     float64
 4   flipper_length_mm  16 non-null     float64
 5   body_mass_g        16 non-null     float64
 6   sex                16 non-null     object 
 7   year               16 non-null     int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 1.1+ KB


<h1> HOW TO FILLUP NULL VALUES</h1>

In [18]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [19]:
penguins['bill_length_mm'].mean()

43.9219298245614

In [20]:
penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())

0      39.10000
1      39.50000
2      40.30000
3      43.92193
4      36.70000
         ...   
339    55.80000
340    43.50000
341    49.60000
342    50.80000
343    50.20000
Name: bill_length_mm, Length: 344, dtype: float64

In [21]:
penguins['bill_length_mm'] =penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())

In [22]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [23]:
penguins['sex'] = penguins['sex'].fillna(penguins['sex'].mode()[0])

In [24]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        2
flipper_length_mm    2
body_mass_g          2
sex                  0
year                 0
dtype: int64

In [25]:
penguins['sex'].mode()

0    male
Name: sex, dtype: object

<h1>How to Change the Data type of a Column</h1>

In [26]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     344 non-null    float64
 3   bill_depth_mm      342 non-null    float64
 4   flipper_length_mm  342 non-null    float64
 5   body_mass_g        342 non-null    float64
 6   sex                344 non-null    object 
 7   year               344 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 21.6+ KB


In [27]:
penguins.columns

Index(['species', 'island', 'bill_length_mm', 'bill_depth_mm',
       'flipper_length_mm', 'body_mass_g', 'sex', 'year'],
      dtype='object')

In [28]:
penguins['mass_kg'] = penguins['body_mass_g'] / 1000

In [29]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000           18.7              181.0   
1       Adelie  Torgersen        39.50000           17.4              186.0   
2       Adelie  Torgersen        40.30000           18.0              195.0   
3       Adelie  Torgersen        43.92193            NaN                NaN   
4       Adelie  Torgersen        36.70000           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream        55.80000           19.8              207.0   
340  Chinstrap      Dream        43.50000           18.1              202.0   
341  Chinstrap      Dream        49.60000           18.2              193.0   
342  Chinstrap      Dream        50.80000           19.0              210.0   
343  Chinstrap      Dream        50.20000           18.7              198.0   

     body_mass_g     sex  year  mass_kg  
0         3750.0    male  2007    3.750  
1         3800.0  female  2007    3.800  
2         3250.0  female  2007    3.250  
3            NaN    male  2007      NaN  
4         3450.0  female  2007    3.450  
..           ...     ...   ...      ...  
339       4000.0    male  2009    4.000  
340       3400.0  female  2009    3.400  
341       3775.0    male  2009    3.775  
342       4100.0    male  2009    4.100  
343       3775.0  female  2009    3.775  

[344 rows x 9 columns]

<h1>Exploratory Data Analytics (EDA)</h1>

<h3>1.What year has the most female record of penguins in Chinstrap species?</h3>

In [30]:
chinstrap_females = penguins[(penguins['species'] == 'Chinstrap') & (penguins['sex'] =='female')]
chinstrap_females['year'].value_counts()

year
2007    13
2009    12
2008     9
Name: count, dtype: int64

<h3>Insight:</h3>

<h4>Based on the data it shows that in the year 2007, it has the most female record of Chinstrap sepcie with a total of 13</h4>

<h3>What is the average body mass of female penguins?</h3>

In [31]:
penguins[penguins['sex'] == 'female']['body_mass_g'].mean()

3862.2727272727275

<h3>Insight</h3>

<h4>According to the data above the average body mass of female penguins is 3862.27 </h4>

<h1>Data Aggregation</h1>

<h2>The use of .groupby()</h2>

In [32]:
penguins_insight8 = penguins.groupby(['species','island','sex','year']).size()

In [33]:
penguins_insight8

species    island     sex     year
Adelie     Biscoe     female  2007     5
                              2008     9
                              2009     8
                      male    2007     5
                              2008     9
                              2009     8
           Dream      female  2007     9
                              2008     8
                              2009    10
                      male    2007    11
                              2008     8
                              2009    10
           Torgersen  female  2007     8
                              2008     8
                              2009     8
                      male    2007    12
                              2008     8
                              2009     8
Chinstrap  Dream      female  2007    13
                              2008     9
                              2009    12
                      male    2007    13
                              2008     9
                      

In [34]:
avg_species_island = penguins.groupby(['species','island'])['body_mass_g'].mean()

In [35]:
avg_species_island

species    island   
Adelie     Biscoe       3709.659091
           Dream        3688.392857
           Torgersen    3706.372549
Chinstrap  Dream        3733.088235
Gentoo     Biscoe       5076.016260
Name: body_mass_g, dtype: float64

In [36]:
mass_species = penguins.groupby(['species', 'body_mass_g']).size()

In [37]:
mass_species

species  body_mass_g
Adelie   2850.0         2
         2900.0         3
         2925.0         1
         2975.0         1
         3000.0         2
                       ..
Gentoo   5850.0         3
         5950.0         2
         6000.0         2
         6050.0         1
         6300.0         1
Length: 136, dtype: int64

<h1>TESTING CORRELATION</h1>

<h3>Pearson R Correlation</h3>
<hr>
<p>Used for testing correlation between two <strong>NUMERICAL VALUES</strong></p>

In [53]:
weight_flipper_cor = penguins['body_mass_g'].corr(penguins['flipper_length_mm'])

In [54]:
weight_flipper_cor

0.8712017673060113

<h1>Numerical-CAtegorical Correlation Test</h1>

<h3>Library-SciPy (Scientific Python)</h3>
<hr>
form scipy.stats import f_oneway

In [55]:
from scipy.stats import f_oneway

In [56]:
x_df = penguins[penguins['sex'] == 'male']

In [57]:
x_df['body_mass_g']

0      3750.0
3         NaN
5      3650.0
7      4675.0
8      3475.0
        ...  
334    3800.0
336    3950.0
339    4000.0
341    3775.0
342    4100.0
Name: body_mass_g, Length: 179, dtype: float64

In [58]:
penguins[penguins['sex'] == 'male']['body_mass_g']

0      3750.0
3         NaN
5      3650.0
7      4675.0
8      3475.0
        ...  
334    3800.0
336    3950.0
339    4000.0
341    3775.0
342    4100.0
Name: body_mass_g, Length: 179, dtype: float64

In [59]:
penguins[penguins['sex'] == 'female']['body_mass_g']

1      3800.0
2      3250.0
4      3450.0
6      3625.0
12     3200.0
        ...  
335    3525.0
337    3650.0
338    3650.0
340    3400.0
343    3775.0
Name: body_mass_g, Length: 165, dtype: float64

In [60]:
f_oneway(penguins[penguins['sex'] == 'male']['body_mass_g'],penguins[penguins['sex'] == 'female']['body_mass_g'])

F_onewayResult(statistic=nan, pvalue=nan)

In [61]:
result,pvalue= f_oneway(penguins[penguins['sex'] == 'male']['body_mass_g'],penguins[penguins['sex'] == 'female']['body_mass_g'])

In [62]:
pvalue

nan

In [63]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        2
flipper_length_mm    2
body_mass_g          2
sex                  0
year                 0
mass_kg              2
dtype: int64